[CI] ProcessWorkerExecutorServiceTests testAutodetectWorkerExecutorService_SubmitAfterShutdown failing

Seems to have been failing 4 times since 23-01-2022

**Build scan:**
https://gradle-enterprise.elastic.co/s/l326w2xhaiwsa/tests/:x-pack:plugin:ml:test/org.elasticsearch.xpack.ml.job.process.ProcessWorkerExecutorServiceTests/testAutodetectWorkerExecutorService_SubmitAfterShutdown

**Reproduction line:**
`./gradlew ':x-pack:plugin:ml:test' --tests "org.elasticsearch.xpack.ml.job.process.ProcessWorkerExecutorServiceTests.testAutodetectWorkerExecutorService_SubmitAfterShutdown" -Dtests.seed=3B1B1ED5C11E48D3 -Dtests.locale=es-CL -Dtests.timezone=America/New_York -Druntime.java=17`

**Applicable branches:**
master, 8.1

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.ml.job.process.ProcessWorkerExecutorServiceTests&tests.test=testAutodetectWorkerExecutorService_SubmitAfterShutdown

**Failure excerpt:**
```
com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=60, name=elasticsearch[AutodetectWorkerExecutorServiceTests][generic][T#1], state=RUNNABLE, group=TGRP-ProcessWorkerExecutorServiceTests]

  at __randomizedtesting.SeedInfo.seed([3B1B1ED5C11E48D3:B9623FFDAA1F91C9]:0)

  Caused by: java.lang.AssertionError: doRun should not be called after the worker is shutdown

    at __randomizedtesting.SeedInfo.seed([3B1B1ED5C11E48D3]:0)
    at org.junit.Assert.fail(Assert.java:88)
    at org.elasticsearch.xpack.ml.job.process.ProcessWorkerExecutorServiceTests$1.doRun(ProcessWorkerExecutorServiceTests.java:59)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:776)
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
    at org.elasticsearch.xpack.ml.job.process.ProcessWorkerExecutorService.start(ProcessWorkerExecutorService.java:115)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:717)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
    at java.lang.Thread.run(Thread.java:833)

```

--- a/docs/changelog/83645.yaml
+++ b/docs/changelog/83645.yaml
@@ -1,0 +1,6 @@
+pr: 83645
+summary: Fix submit after shutdown in process worker service
+area: Machine Learning
+type: bug
+issues:
+ - 83633

--- a/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/ProcessWorkerExecutorService.java
+++ b/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/ProcessWorkerExecutorService.java
@@ -95,6 +95,7 @@ public class ProcessWorkerExecutorService extends AbstractExecutorService {
             EsRejectedExecutionException rejected = new EsRejectedExecutionException(processName + " worker service has shutdown", true);
             if (command instanceof AbstractRunnable runnable) {
                 runnable.onRejection(rejected);
+                return;
             } else {
                 throw rejected;
             }

